### PR TITLE
Add/unify decentlab devices

### PR DESCRIPTION
**Pull request by decentlab**
## Decoder Description

Added all Decentlab device (connector) decoders from TTN repository.

## Type of change

- [x] Adding a new Decoder of Connector
- [x] Update or fixing an issue in existing Decoder

## Decoder Information and Payload to test and review

- [x] Documentation of the hardware or protocol: https://www.decentlab.com/support
- [x] Payload of example the test the decoder: https://github.com/decentlab/decentlab-decoders

## Checklist for Adding a New Decoder

- [x] Added a `network.jsonc` or `connector.jsonc` file that follows the structure defined in `./schema/`.
- [x] Created a new folder under `./decoders/network/` or `./decoders/connector/` with the name of your decoder.
- [x] Created version folders and added `manifest.jsonc` files for each version.
- [x] Followed the folder structure guidelines for manufacturer and sensor/device model.
- [x] The code has unit test and it's in TypeScript.

## Additional Notes

The existing decoders have been reverted back to the original TTN device repo version. This simplifies creating the decoders and makes sure that they uniformly work as expected over many platforms. Besides, the modifications made by TagoIO was mainly cosmetic, and there was no logic change.

We intentionally keep the decoders' syntax such that they're supported on most platforms with a minimal change.

### CODE REVIEW

**Review by mateuscardosodeveloper — CHANGES_REQUESTED** [review]
There are a few points that need to be corrected or improved:

1. The images should have a transparent background.
2. The `description.md` file serves as the subtitle displayed on the devices' screen.
   a. This field is searchable. (When a user searches for a device on the platform, they use these words).
   b. It is recommended that `description.md` be clear and focus on the specific capabilities and features of the device.
3. 3. The new decoders added must be made in TypeScript and also add type annotation. E.g:
```ts
function toTagoFormat(object_item: {}, group: number, prefix: string = "") {
  //...
}
```
  a. When changing the payload to TypeScript, don't forget to edit the `connector.jsonc` file to `payload.ts`.

When changing the payload to TypeScript, don't forget to edit the `connector.jsonc` file to `payload.ts`.
4. Each decoder must have unit tests. Here is an example of a unit test you can use.
```ts
import { describe, test, expect } from "vitest";

import { decoderRun } from "../../../../../src/functions/decoder-run";

const file_path = "decoders/connector/your-path/payload.js" as const;

describe("Shall not be parsed", () => {
  let payload = [{ variable: "shallnotpass", value: "04096113950292" }];
  payload = decoderRun(file_path, { payload });
  test("Output Result", () => {
    expect(Array.isArray(payload)).toBe(true);
  });

  test("Not parsed Result", () => {
    expect(payload).toEqual([{ variable: "shallnotpass", value: "04096113950292" }]);
  });
});
```

If you can add a unit test for the payload parser, that would be great.

**Comment by decentlab** [comment]
Thanks for the review.

We followed the existing payload decoders that were added by TagoIO. It would have been more efficient if we knew the different requirements before creating these decoders. Would it be possible to add the decoders now as it is, and we improve them iteratively?

Some questions:

- 2a, 2b: Could you make an example improvement on one device please?
- 3a: Do you mean that we need to restructure all payloads to a real TypeScript? Or just type annotations on the functions?  Or change just the `toTagoFormat` signature?

**Comment by mateuscardosodeveloper** [comment]
> Thanks for the review.
> 
> We followed the existing payload decoders that were added by TagoIO. It would have been more efficient if we knew the different requirements before creating these decoders. Would it be possible to add the decoders now as it is, and we improve them iteratively?
> 
> Some questions:
> 
> * 2a, 2b: Could you make an example improvement on one device please?
> * 3a: Do you mean that we need to restructure all payload decoders to a real TypeScript? Or just add type annotations in the function signatures? Or change just the `toTagoFormat` signature?

Since the payload was created by TagoIO, I see no issue in approving it as it currently stands. However, I kindly request that you add the unit test mentioned in topic 4.

Regarding points 2a and 2b: These are just informational, indicating that they pertain to the searchable fields within the platform for the sensor.
searchable-devices

**Comment by decentlab** [comment]
Thanks for considering to approve the decoders as they are now. We have added unit tests for the payload parsers.

File: decoders/connector/decentlab/dl-alb/description.md [hunk 1]
```diff
@@ -0,0 +1 @@
+DL-ALB is an albedometer sensor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-alb/description.md:1]: Please remove the sensor name, example: "Albedometer sensor over LoRaWAN®"

File: decoders/connector/decentlab/dl-dlr2-010/description.md [hunk 2]
```diff
@@ -0,0 +1 @@
+DL-DLR2-010 is a dual pulse counter dry contact transmitter over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-dlr2-010/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-dlr2-012/description.md [hunk 3]
```diff
@@ -0,0 +1 @@
+DL-DLR2-012 is an analog strain gauge sensor transmitter over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-dlr2-012/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-dlr2-011/description.md [hunk 4]
```diff
@@ -0,0 +1 @@
+DL-DLR2-011 is a dual digital input dry contact transmitter over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-dlr2-011/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-ds18/description.md [hunk 5]
```diff
@@ -0,0 +1 @@
+DL-DS18 is a temperature sensor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-ds18/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-dws/description.md [hunk 6]
```diff
@@ -0,0 +1 @@
+DL-DWS is a weighing scale over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-dws/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-gmm/description.md [hunk 7]
```diff
@@ -0,0 +1 @@
+DL-GMM is a greenhouse multi monitor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-gmm/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-iam/description.md [hunk 8]
```diff
@@ -1 +1 @@
-DL-IAM is a Indoor ambiance monitor including co2, tvoc and motion sensor over LoRaWAN®
\ No newline at end of file
+DL-IAM is an indoor ambiance monitor including co₂, tvoc and motion sensor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-iam/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-isf/description.md [hunk 9]
```diff
@@ -0,0 +1 @@
+DL-ISF is a sapflow sensor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-isf/description.md:1]: Please remove sensor name.

File: decoders/connector/decentlab/dl-itst/description.md [hunk 10]
```diff
@@ -0,0 +1 @@
+DL-ITST is an infrared thermometer/surface temperature sensor over LoRaWAN®
```
Comment (Freddyminu) [inline decoders/connector/decentlab/dl-itst/description.md:1]: Please remove sensor name.

**Review by Freddyminu — COMMENTED** [review]
Please remove the sensor name from all **description.md** files.

Additionally, ensure that the **install_text** contains only factual information about the device, avoiding any subjective phrases.

**Comment by decentlab** [comment]
@Freddyminu, these have been done by TagoIO and we followed exactly how it was. Would it be possible if you approve this PR now, and we improve iteratively? Thanks.

**Review by Freddyminu — APPROVED** [review]

**Review by mateuscardosodeveloper — APPROVED** [review]

**Pull request merged**